File: src/main/java/work/lclpnet/ap2/api/game/data/DataEntry.java
package work.lclpnet.ap2.api.game.data;

import org.jetbrains.annotations.Nullable;
import work.lclpnet.kibu.translate.TranslationService;
import work.lclpnet.kibu.translate.text.TranslatedText;

public interface DataEntry<Ref extends SubjectRef> {

    Ref subject();

    @Nullable
    TranslatedText toText(TranslationService translationService);
}

File: src/lib/src/main/java/work/lclpnet/ap2/api/game/data/DataEntry.kt
package work.lclpnet.ap2.api.game.data

import work.lclpnet.kibu.translate.Translations
import work.lclpnet.kibu.translate.text.TranslatedText

interface DataEntry<Ref : SubjectRef> {

    val subject: Ref

    fun toText(translationService: Translations): TranslatedText?

    fun scoreEquals(other: DataEntry<Ref>): Boolean
}
